test/NuGet.Core.Tests/NuGet.DependencyResolver.Core.Tests/RemoteDependencyWalkerTests.cs

```diff
@@ -823,176 +818,246 @@ public void IsGreaterThanEqualTo_ReturnsFalse_IfRightVersionIsLargerThanLeft(str
+            // A ~> D (the version 2.0.0 or as defined by "centralPackageVersion" argument
+            //         the dependency is not direct,
+            //         it simulates the fact that there is a centrally defined "D" package
+            //         the information is added to the provider)
+
+            provider.Package("A", otherVersion)
+                    .DependsOn("B", otherVersion);
+
+            provider.Package("B", otherVersion)
+                   .DependsOn("C", otherVersion);
+
+            provider.Package("C", otherVersion)
+                  .DependsOn(centralPackageName, otherVersion);
+
+            // Simulates the existence of a D centrally defined package that is not direct dependency
+            provider.Package("A", otherVersion)
+                     .DependsOn(centralPackageName, centralPackageVersion, LibraryDependencyTarget.Package, versionCentrallyManaged: true);
+
+            // Add central package to the source with multiple versions
+            provider.Package(centralPackageName, "1.0.0");
+            provider.Package(centralPackageName, centralPackageVersion);
+            provider.Package(centralPackageName, "3.0.0");
+
+            context.LocalLibraryProviders.Add(provider);
+            var walker = new RemoteDependencyWalker(context);
+
+            // Act
+            var rootNode = await DoWalkAsync(walker, "A", framework);
+
+            var analyzeResult = rootNode.Analyze();
 
-            public Task<LibraryDependencyInfo> GetDependenciesAsync(
-                LibraryIdentity match,
-                NuGetFramework targetFramework,
-                SourceCacheContext cacheContext,
-                ILogger logger,
-                CancellationToken cancellationToken)
+            // Assert
+            Assert.Equal(1, analyzeResult.Downgrades.Count);
+            var downgrade = analyzeResult.Downgrades.First();
+            Assert.Equal(centralPackageName, downgrade.DowngradedFrom.Key.Name);
+            Assert.Equal("[3.0.0, )", downgrade.DowngradedFrom.Key.VersionRange.ToNormalizedString());
+            Assert.Equal(centralPackageName, downgrade.DowngradedTo.Key.Name);
+            Assert.Equal("[2.0.0, )", downgrade.DowngradedTo.Key.VersionRange.ToNormalizedString());
+        }
+
+        [Theory]
+        [InlineData(LibraryDependencyReferenceType.Direct, true)]
+        [InlineData(LibraryDependencyReferenceType.Transitve, true)]
+        [InlineData(LibraryDependencyReferenceType.None, true)]
+        [InlineData(LibraryDependencyReferenceType.Direct, false)]
+        [InlineData(LibraryDependencyReferenceType.Transitve, false)]
+        [InlineData(LibraryDependencyReferenceType.None, false)]
+        public void IsDependencyValidForGraphTest(LibraryDependencyReferenceType referenceType, bool versionCentrallyManaged)
+        {
+            var centralPackageName = "D";
+            var framework = NuGetFramework.Parse("net45");
```

Comment:
Looks like this variable is not used in this test.

Reply:
Removed.